.buildkite/scripts/build_pr.sh

```diff
@@ -103,14 +103,19 @@ if [[ "${GITHUB_PR_BASE_REPO}" != 'docs' ]]; then
       docs_diff=$(git diff --stat "origin/$GITHUB_PR_TARGET_BRANCH"...HEAD -- ./docs/asciidoc)
       ;;
 
-    "eland" | "enterprise-search-php" | "enterprise-search-python" | "enterprise-search-ruby")
+    "eland")
+      git fetch origin "$GITHUB_PR_TARGET_BRANCH"
+      docs_diff=$(git diff --stat "origin/$GITHUB_PR_TARGET_BRANCH"...HEAD -- "./docs/guide/**/*.asciidoc")
+      ;;
+
+    "enterprise-search-php" | "enterprise-search-python" | "enterprise-search-ruby")
       git fetch origin "$GITHUB_PR_TARGET_BRANCH"
       docs_diff=$(git diff --stat "origin/$GITHUB_PR_TARGET_BRANCH"...HEAD -- ./docs/guide)
       ;;
 
     "elasticsearch")
       git fetch origin "$GITHUB_PR_TARGET_BRANCH"
-      docs_diff=$(git diff --stat "origin/$GITHUB_PR_TARGET_BRANCH"...HEAD -- ./buildSrc ./build-tools-internal ./build-tools/src/main/resources ./client ./docs ./modules/reindex/src/internalClusterTest/java/org/elasticsearch/client/documentation ./modules/reindex/src/test/java/org/elasticsearch/client/documentation ./plugins/examples ./server/src/internalClusterTest/java/org/elasticsearch/client/documentation ./server/src/main/resources/org/elasticsearch/common ./server/src/test/java/org/elasticsearch/client/documentation ./x-pack/docs ./x-pack/plugin/esql/qa/testFixtures/src/main/resources ./x-pack/plugin/sql/qa ./x-pack/qa/sql)
+      docs_diff=$(git diff --stat "origin/$GITHUB_PR_TARGET_BRANCH"...HEAD -- ./buildSrc ./build-tools-internal ./build-tools/src/main/resources ./client "./docs/**/*.asciidoc" ./modules/reindex/src/internalClusterTest/java/org/elasticsearch/client/documentation ./modules/reindex/src/test/java/org/elasticsearch/client/documentation ./plugins/examples ./server/src/internalClusterTest/java/org/elasticsearch/client/documentation ./server/src/main/resources/org/elasticsearch/common ./server/src/test/java/org/elasticsearch/client/documentation ./x-pack/docs ./x-pack/plugin/esql/qa/testFixtures/src/main/resources ./x-pack/plugin/sql/qa ./x-pack/qa/sql)
```

Comment:
I'm noticing the same thing here. `**` appears to only match one or more directly. This means changes to files in the root of the dir directly before `**` is used will not correctly indicate changes. For example:

If you update in the es repo `docs/versions.asciidoc`, it will not indicate a change with

```
 git diff --name-only  -- "./docs/**/*.asciidoc"
```

Reply:
Wow, TIL a lot about pattern matching in git commands. From the git docs:

<img width="712" alt="Screenshot 2025-03-14 at 1 31 12 PM" src="https://github.com/user-attachments/assets/54f2bc10-7547-4010-9afa-c0dc49059e3a" />

That's why using a pattern like `./docs/**/*.asciidoc`, matches `.asciidoc` files in subdirectories of `docs/`, but not directly within `docs/`. The pattern expects at least one directory level after `docs/` before matching `.asciidoc` files.

I think we can include `docs/` as well as its subdirectories with `./docs/**/**/*.asciidoc`. Testing that now.

Reply:
Okay that didn't work but what does work is `"./docs/**.asciidoc"`

Reply:
```suggestion
      docs_diff=$(git diff --stat "origin/$GITHUB_PR_TARGET_BRANCH"...HEAD -- ./buildSrc ./build-tools-internal ./build-tools/src/main/resources ./client "./docs/**.asciidoc" ./modules/reindex/src/internalClusterTest/java/org/elasticsearch/client/documentation ./modules/reindex/src/test/java/org/elasticsearch/client/documentation ./plugins/examples ./server/src/internalClusterTest/java/org/elasticsearch/client/documentation ./server/src/main/resources/org/elasticsearch/common ./server/src/test/java/org/elasticsearch/client/documentation ./x-pack/docs ./x-pack/plugin/esql/qa/testFixtures/src/main/resources ./x-pack/plugin/sql/qa ./x-pack/qa/sql)
```